planet/clients/orders.py

```diff
@@ -302,68 +302,51 @@ async def aggregated_order_stats(self) -> dict:
         resp = await self._do_request(req)
         return resp.json()
 
-    async def download_asset(self,
-                             location: str,
-                             filename: str = None,
-                             directory: str = None,
+    async def download_order(self,
+                             order: Order,
+                             directory: Path,
                              overwrite: bool = False,
-                             progress_bar: bool = True) -> str:
-        """Download ordered asset.
+                             callback=None,
+                             callback_args: tuple = None,
+                             callback_kwargs: dict = None) -> None:
+        """Download all assets in a finished order.
 
         Parameters:
-            location: Download location url including download token.
-            filename: Custom name to assign to downloaded file.
-            directory: Write to given directory instead of current directory.
+            order: a finished Order.
+            directory: a download directory Path.
             overwrite: Overwrite any existing files.
-            progress_bar: Show progress bar during download.
 
         Returns:
-            Path to downloaded file.
+            None.
 
         Raises:
-            planet.exceptions.APIException: On API error.
-        """
-        req = self._request(location, method='GET')
-
-        async with self._session.stream(req) as resp:
-            body = StreamingBody(resp)
-            dl_path = os.path.join(directory or '.', filename or body.name)
-            await body.write(dl_path,
-                             overwrite=overwrite,
-                             progress_bar=progress_bar)
-        return dl_path
+            OrderError
 
-    async def download_order(self,
-                             order_id: str,
-                             directory: str = None,
-                             overwrite: bool = False,
-                             progress_bar: bool = False) -> typing.List[str]:
-        """Download all assets in an order.
-
-        Parameters:
-            order_id: The ID of the order
-            directory: Write to given directory instead of current directory.
-            overwrite: Overwrite any existing files.
-            progress_bar: Show progress bar during download.
+        """
+        semaphore = asyncio.Semaphore(MAX_CONCURRENCY)
```

Comment:
asyncio.Semaphore allows us to put a cap on the number of simultaneous downloads.